### PR TITLE
feat(AV-1694): apiset managed datasets

### PR DESCRIPTION
**Pull request by teravill**
Datasets can now be associated with an apiset via manage datasets. Also added the new, read, edit and resources routes to the plugin, making overloading them easier. 

image
image

### CODE REVIEW

File: ckanext/apis/helpers.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import ckan.model as model
+from ckan.plugins import toolkit as tk
+
+c = tk.c
```
Comment (beazil) [inline ckanext/apis/helpers.py]: I think tk.c was deprecated, and it's recommended to use tk.g nowadays (https://docs.ckan.org/en/2.9/extensions/plugins-toolkit.html#ckan.plugins.toolkit.ckan.plugins.toolkit.c)

File: ckanext/apis/logic/auth.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        toolkit.check_access('package_create', context, data_dict)
+        return {'success': True}
+    except toolkit.NotAuthorized:
+        return {'success': False,
+                'msg': toolkit._('User {0} not authorized to create apisets').format(user)}
+
+def apiset_association_create(context, data_dict):
+    user = context.get('user')
+    try:
+        toolkit.check_access('package_create', context, data_dict)
```
Comment (beazil) [inline ckanext/apis/logic/auth.py]: Should this actually check update access (I think that associating something for apiset is more like an update than create)

**Review by beazil — CHANGES_REQUESTED** [review]

File: ckanext/apis/helpers.py (hunk 1 above)
Comment (teravill) [reply ckanext/apis/helpers.py]: Implemented in [48e44f1](https://github.com/vrk-kpa/ckanext-apis/pull/7/commits/48e44f1abb34844a2b5579e0de5e99e033c81dd7)

File: ckanext/apis/logic/auth.py (hunk 2 above)
Comment (teravill) [reply ckanext/apis/logic/auth.py]: Implemented in [48e44f1](https://github.com/vrk-kpa/ckanext-apis/pull/7/commits/48e44f1abb34844a2b5579e0de5e99e033c81dd7)

**Review by beazil — APPROVED** [review]

**Pull request merged**